### ISSUE TITLE
Can't connect to amqp server [rabbit]

### ISSUE THREAD

**Issue by delkirksey** [omniti-labs/pg_amqp#24]
I installed this plugin in a Docker container that holds my Postgres instance only. I am attempting to publish messages to a RabbitMQ broker housed in another container on the same network. 

I am able to communicate from an Airflow instance in another container to the RabbitMQ server with ease. I am able to see the RabbitMQ container and the open ports from within the Postgres container using: 

```
root@**********:~/pg_amqp# nc -v rabbit 5672
airflow-postgres-rabbit.airflow-postgres-default
rabbit [172.*.*.*] 5672 (amqp) open
```

My broker table is as follows (username & password confirmed):

```
postgres=# select * from amqp.broker;
 broker_id |  host  | port | vhost | username | password 
-----------+--------+------+-------+----------+----------
         1 | rabbit | 5672 | /     | admin    | mypass
```

When running the publish command, I see this:

```
postgres=# SELECT amqp.publish(1, 'amq.direct', 'test', 'test');
WARNING:  amqp[rabbit:5672] login failed on broker 1
 publish 
---------
 f
(1 row)
```

I have tried substituting the hostname for the IP address of the rabbit server.
I have been substituting in different values for the `exchange` including `default` and `amqp.direct`.
I have been substituting in a different value for the queue including `default`.
I have been substituting in a different value for the message as a prayer.

I'm sure I goofed up somewhere so any help is appreciated. Willing to provide as much additional context as needed.

**Comment by hlobit** [comment]
It is caused by the new `max-channels` limit of recent rabbitmq versions: https://github.com/omniti-labs/pg_amqp/issues/23

I applied the given patch to resolve the issue.

FYI, you could check, for me the issue was reported in rabbitmq container logs.

```
rabbitmq_1  | 2019-01-15 13:10:47.249 [info] <0.614.0> accepting AMQP connection <0.614.0> (192.168.160.3:57946 -> 192.168.160.2:5672)                                                                               
db_1        | 2019-01-15 13:10:49.274 UTC [137] WARNING:  amqp[rabbitmq:5672] login failed on broker 1                                                                                                               
rabbitmq_1  | 2019-01-15 13:10:50.258 [error] <0.614.0> closing AMQP connection <0.614.0> (192.168.160.3:57946 -> 192.168.160.2:5672):                                                                               
rabbitmq_1  | failed to negotiate connection parameters: negotiated channel_max = 0 (no limit) is higher than the maximum allowed value (2047)
```

**Comment by xzilla** [comment]
Per comment by @hlobit, I believe this has now been addressed in pr #26.
